### DISCUSSION TITLE
Any plan on HiDPI support?

### DISCUSSION THREAD

**Discussion by felixding** [trunkmaster/nextspace#381] [Q&A]
Just installed Nextspace and found no HiDPI support out of the box. Does it support HiDPI? Thank you for this wonderful project.

**Comment by trunkmaster** [comment]
There is no support for HiDPI yet. However I plan to add HiDPI support.

**Comment by felixding** [comment]
What does it take to add HiDPI support? Can't wait to switch to Nextspace. Don't know how to code in C but would like to help.

**Comment by uchuugaka** [comment]
In some cases it can be as simple as X config, but can also include things like additional image assets for UI rendered by image assets and if anything is drawn with code, it needs to account for that also.
There are also multiple ways to work with HiDPI (or any DPI). 
You can basically aim for more detail at a particular physical pixel size and pitch (space between pixels) or you can aim toward fitting more things onto a screen.
And sometimes you have multiple displays with different specs at the same time.
It's not terribly impossible but plenty of non trivial things.
Testing would probably be a great thing to provide.

**Comment by jafd** [comment]
If I may have a suggestion on how to proceed with transition to HiDPI while not all components are ready for it: draw everything at 2x size with nearest-neighbor interpolation. This will make them pixelated, true, but not tiny. GTK+/Qt applications in such environment will look fine, since they "know" how to handle it already (provided some magical environment variables are in place, and/or Gsettings daemon running).

Every system in existence which had to enlarge legacy applications so they were not tiny on HiDPI screens, resorted to some kind of antialiasing which made them look awfully blurry and unreadable. Yes, even OS X/macOS in the early days of Retina displays drew older applications in this blurry way.

The only platform which did the right thing was VirtualBox: when you had to run a low-DPI OS on a high-DPI screen, you could select the size to be 200% or 300%, and it would just double or triple pixels, resulting in sharp image.
